core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -520,27 +521,28 @@ private class LiveRDDDistribution(exec: LiveExecutor) {
 
 }
 
-private class LiveRDD(val info: RDDInfo) extends LiveEntity {
+/**
+ * Tracker for data related to a persisted RDD. Note the storage level is kept immutable here,
+ * following the current behavior of `RDD.persist()`, even though it is mutable in the `RDDInfo`
+ * structure.
+ */
```

Comment:
A stage may have been submitted before the RDD was persisted at all, then another stage submitted after the RDD is persisted, so its not actually immutable.  You wouldn't properly capture that here.

(probably not the optimal thing for the user to do, but I've seen weirder things ...)

Reply:
Actually that works fine, because the listener does not track RDDs that are not persisted; so there wouldn't be a live RDD for the first stage in your example; it would be created when the second stage is submitted, and at that point the storage level cannot be changed further.

I'll update the comment (maybe even add a unit test).

Reply:
makes sense, thanks for adding the test too.